standard/ERCs/erc-6900.md

```diff
@@ -537,36 +526,51 @@ Finally, the function MUST emit the event `ModuleUninstalled` with the module's
+
+The account MUST define a way to pass data separately for each validation hook and the validation function itself. This data MUST be sent as the `userOp.signature` field for user op validation, the `authorization` field for runtime validation, and the `signature` field for signature validation.
+
+The result of user op validation SHOULD be the intersection of time bounds returned by the pre validation hooks and the validation function. If any validation hooks or the validation functions returns a value of `1` for the authorizer field, indicating a signature verification failure by the ERC-4337 standard, the account MUST return a value of `1` for the authorizer portion of the validation data.
+
+The set of validation hooks run MUST be the hooks specified by account state at the start of validation. In other words, if the set of applicable hooks changes during validation, the original set of hooks MUST still run, and only future invocations of the same validation should reflect the changed set of hooks.
+
+#### Checking Validation Applicability
+
+To enforce module permission isolation, the modular account MUST check validation function applicability as part of each validation function implementation.
+
+User op validation and runtime validation functions have a configurable range of applicability to functions on the account. This can be configured with selectors installed to a validation. Alternatively, a validation installation may specify the `isGlobal` flag as true, which means the account MUST consider it applicable to any module execution function with the `allowGlobalValidation` flag set to true, or for any account native function that the account MAY allow for global validation.
 
-When the function `validateUserOp` is called on modular account by the `EntryPoint`, it MUST find the user operation validation function associated to the function selector in the first four bytes of `userOp.callData`. If there is no function defined for the selector, or if `userOp.callData.length < 4`, then execution MUST revert.
+If the selector being checked is `execute` or `executeBatch`, the modular account MUST perform additional checking. If the target of `execute` is the modular account's own address, validation MUST revert, to prevent privilege escalation. If the target of any `Call` within `executeBatch` is the account, the selector of the `Call` MUST also be checked to apply for the given validation. If the batch self-call's selector is again either `execute` or `executeBatch`, validation MUST revert, to prevent privilege escalation without requiring arbitrary-depth calldata decoding.
```

Comment:
Should we be flexible on this?

Reply:
Hm, it's part of the "validation applicability checking" logic that is important for enforcing module security. But I suppose we don't have any strict requirements on how that checking is implemented. The standard could allow for accounts to either do arbitrary-depth calldata decoding, or to just force reverts on depth > 1.

Maybe we reword this just to require checking validation applicability to any self-calls or batching, but remove the requirements for:
- "revert on `execute` towards self"
- "revert on nested `executeBatch` towards self" 

wdyt?

Reply:
Sounds good.

Reply:
Updated